lib/types/ipblock.go

```diff
@@ -0,0 +1,173 @@
+package types
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"net"
+	"strings"
+)
+
+// ipBlock represents a continuous segment of IP addresses
+type ipBlock struct {
+	// TODO rename count as it is actually the accumulation of the counts up to and including this
+	// on in the pool
+	// maybe add another field, although technically we need count only to find out when to enter this
+	// block not for anything else
+	firstIP, count *big.Int
+	ipv6           bool
+}
+
+// IPPool represent a slice of IPBlocks
+type IPPool struct {
+	list  []ipBlock
+	count *big.Int
+}
+
+func getIPBlock(s string) (*ipBlock, error) {
+	switch {
+	case strings.Contains(s, "-"):
+		return ipBlockFromRange(s)
+	case strings.Contains(s, "/"):
+		return ipBlockFromCIDR(s)
+	default:
+		if net.ParseIP(s) == nil {
+			return nil, fmt.Errorf("%s is not a valid ip, range ip or CIDR", s)
+		}
+		return ipBlockFromRange(s + "-" + s)
+	}
+}
+
+func ipBlockFromRange(s string) (*ipBlock, error) {
+	ss := strings.SplitN(s, "-", 2)
+	ip0Str, ip1Str := strings.TrimSpace(ss[0]), strings.TrimSpace(ss[1])
+	ip0, ip1 := net.ParseIP(ip0Str), net.ParseIP(ip1Str)
+	if ip0 == nil || ip1 == nil {
+		fmt.Println("Wrong IP range format: ", s)
```

Comment:
huh, completely missed that, but do we need this at all? seems like a forgotten debug statement, we're returning an error below

Reply:
This will be deleted (already did it locally) - the original code didn't return errors it was printing them. 
I apparently missed some :facepalm:  `fmt.Println`